docs/BuildFromSource.md

```diff
@@ -1,155 +1,201 @@
+NodeJS installes the Node package manager (npm) by default. This repo depends on yarn, an alternate package manager for the Node ecosystem. You can install Yarn from the command line using the following command.
 
-### Common error: Unable to locate the .NET Core SDK
+```bash
+$ npm install -g yarn
+```
 
-Executing `.\restore.cmd` or `.\build.cmd` may produce these errors:
+#### Java Development Kit
 
-> error : Unable to locate the .NET Core SDK. Check that it is installed and that the version specified in global.json (if any) matches the installed version.
-> error MSB4236: The SDK 'Microsoft.NET.Sdk' specified could not be found.
+This repo contains some Java source code that depends on an install of the JDK v11 or newer. The JDK can be installed from either:
+  * OpenJDK <https://jdk.java.net/>
+  * Oracle's JDK <https://www.oracle.com/technetwork/java/javase/downloads/index.html>
 
-In most cases, this is because the option _Use previews of the .NET Core SDK_ in VS2019 is not checked. Start Visual Studio, go to _Tools > Options_ and check _Use previews of the .NET Core SDK_ under _Environment > Preview Features_.
+Similar to [the instructions above for Windows](#java-development-kit-in-windows), be sure that the the `JAVA_HOME` environment variable is set to the location of your Java installation.
 
-### Common error: HTTP Error 500.33 - ANCM Request Handler Load Failure
+## Step 3: Build the repo
 
-The [ASP.NET Core Module](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/aspnet-core-module) (ANCM) for IIS is not supported when running projects in this repository.
+Before opening our .sln/.slnf files in Visual Studio or VS Code, you will need to build the repo locally.
 
-After using `startvs.cmd` to open a solution in Visual Studio, the Kestrel web host option must be used (name of the project) and not IIS Express.
+### In Visual Studio
 
-Example of running the `MvcSandbox` project:
+To set up your project for development on Visual Studio, you'll need to execute the following command.
 
-`.\startvs.cmd .\src\Mvc\Mvc.sln`
+```ps1
+PS1> .\restore.cmd
+```
 
-![Web host options in Visual Studio](./vs-iis-express-aspnet-core-mvc-sandbox.jpg)
+> :bulb: If you happen to be working on macOS or Linux, you can use the `restore.sh` command.
+
+This will download the required tools and restore all projects inside the repository. At that point, you should be able
+to open the .sln file or one of the project specific .slnf files to work on the projects you care about.
+
+   > :bulb: Pro tip: you will also want to run this command after pulling large sets of changes. On the master
+   > branch, we regularly update the versions of .NET Core SDK required to build the repo.
+   > You will need to restart Visual Studio every time we update the .NET Core SDK.
+
+To open the solution file in Visual Studio, run the following command.
+
+```ps1
+PS1> .\startvs.cmd
```

Comment:
I don't think it does - the root `startvs.cmd` makes you name a project, so it only does this if you say `.\startvs.cmd AspNetCore.sln`.

Reply:
Fixed.